prediction_market_agent_tooling/deploy/betting_strategy.py

```diff
@@ -188,55 +228,83 @@ def calculate_trades(
+            # use Kelly simple, since Kelly full only supports 2 outcomes
+            kelly_bet = get_kelly_bet_simplified(
+                max_bet=market.get_usd_in_token(self.max_bet_amount),
+                market_p_yes=market.probabilities[direction],
+                estimated_p_yes=answer.probabilities[direction],
+                confidence=answer.confidence,
+            )
+        else:
+            # We consider only binary markets, since the Kelly strategy is not yet implemented
+            # for markets with more than 2 outcomes (https://github.com/gnosis/prediction-market-agent-tooling/issues/671).
+            direction_to_bet_pool_size = outcome_token_pool[direction]
+            other_direction_pool_size = outcome_token_pool[other_direction]
+            kelly_bet = get_kelly_bet_full(
+                yes_outcome_pool_size=direction_to_bet_pool_size,
+                no_outcome_pool_size=other_direction_pool_size,
+                estimated_p_yes=answer.probabilities[direction],
+                max_bet=market.get_usd_in_token(self.max_bet_amount),
+                confidence=answer.confidence,
+                fees=market.fees,
+            )
 
         kelly_bet_size = kelly_bet.size
         if self.max_price_impact:
             # Adjust amount
             max_price_impact_bet_amount = self.calculate_bet_amount_for_price_impact(
-                market, kelly_bet
+                market, kelly_bet, direction=direction
             )
 
             # We just don't want Kelly size to extrapolate price_impact - hence we take the min.
             kelly_bet_size = min(kelly_bet.size, max_price_impact_bet_amount)
 
+        bet_outcome = direction if kelly_bet.direction else other_direction
         amounts = {
-            market.get_outcome_str_from_bool(
-                kelly_bet.direction
-            ): market.get_token_in_usd(kelly_bet_size),
+            bet_outcome: market.get_token_in_usd(kelly_bet_size),
         }
         target_position = Position(market_id=market.id, amounts_current=amounts)
         trades = self._build_rebalance_trades_from_positions(
             existing_position, target_position, market=market
         )
         return trades
 
+    def get_outcome_prices_from_balances(self, balances: list[float]) -> list[float]:
+        num_balances = len(balances)
+        prices = []
+
+        # Compute the product of all outcome balances except i for each i
+        for i in range(num_balances):
+            numerator = prod([balances[j] for j in range(num_balances) if j != i])
+            prices.append(numerator)
+
+        denominator = sum(prices)
+        normalized_prices = [p / denominator for p in prices]
+
+        return normalized_prices
+
     def calculate_price_impact_for_bet_amount(
         self,
-        buy_direction: bool,
+        outcome_idx: int,
         bet_amount: CollateralToken,
-        yes: OutcomeToken,
-        no: OutcomeToken,
+        pool_balances: list[float],
```

Comment:
```suggestion
        pool_balances: list[OutcomeToken],
```

then you don't need to cast it down the line and it's type safe

Reply:
Fixed types here https://github.com/gnosis/prediction-market-agent-tooling/pull/666/commits/8070f79ce437b0dc449bd45ebdfa419ecff552a7